repositories/strategies/routingStrategy.js

```diff
@@ -0,0 +1,127 @@
+const { head } = require("lodash/fp");
+const { concat, parseInt } = require("lodash");
+
+const updateConditionsTable = async (trx, { id, answerId }) =>
+  trx("Routing_Conditions")
+    .where({ id })
+    .update({
+      AnswerId: answerId
+    })
+    .returning("*")
+    .then(head);
+
+const updateValuesTable = async (trx, { id }) =>
+  trx("Routing_ConditionValues")
+    .where({ ConditionId: id })
+    .del()
+    .returning("*")
+    .then(head);
+
+const getConditionAnswerId = async (trx, { id }) =>
+  trx("Routing_Conditions")
+    .select("AnswerId")
+    .where({ id })
+    .then(head);
+
+const getSectionId = async (trx, id) =>
+  trx("Pages")
+    .select("SectionId", "order")
+    .where({ id })
+    .then(head);
+
+const getQuestionnaireId = async (trx, { SectionId }) =>
+  trx("Sections")
+    .select("QuestionnaireId", "id")
+    .where({ id: SectionId })
+    .then(head);
+
+const getDestinationsInSection = async (trx, { SectionId, order }) =>
+  trx("Pages")
+    .select("id")
+    .where({ SectionId })
+    .where("order", ">", order);
+
+const getSectionsAheadOfPage = async (
+  trx,
+  { QuestionnaireId },
+  { SectionId }
+) =>
+  trx("Sections")
+    .select("id")
+    .where({ QuestionnaireId })
+    .where("id", ">", SectionId);
+
+const getDestinationsOutsideSection = async (trx, sectionIds) => {
+  const idArray = sectionIds.map(sectionId => sectionId.id);
+  return trx("Pages")
+    .select("id")
+    .from(function() {
+      this.select("SectionId")
+        .from("Pages")
+        .min("order as min")
+        .whereIn("SectionId", idArray)
+        .groupBy("SectionId")
+        .as("firstPages");
+    })
+    .innerJoin("Pages", function() {
+      this.on("firstPages.SectionId", "=", "Pages.SectionId").andOn(
+        "firstPages.min",
+        "=",
+        "Pages.order"
+      );
+    });
+};
+
+module.exports.updateConditions = async function updateConditions(
+  trx,
+  { id, answerId }
+) {
+  const prevAnswerId = await getConditionAnswerId(trx, { id });
+  if (parseInt(answerId) === prevAnswerId.AnswerId) {
+    return;
```

Comment:
What's the scenario where this would be called? Would we ever fire an update from the UI if the answer doesn't change?

Reply:
I wasn't sure, does seem unlikely that there would be an update if nothing changes. Just felt a little warranted as we're doing hard deletes here?